### PR TITLE
Adding messaging tab

### PR DESCRIPTION
**Pull request by ravjain-adb**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Swift/AEPSampleApp/MessagingView.swift [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+            TextField("", text: $personalisedData)
+                .textFieldStyle(RoundedBorderTextFieldStyle())
+                .autocapitalization(.none)
+
+            Button(action: {
+                self.sendProfileData()
+            }){
+                HStack {
+                    Text("Update Name in AEP profile")
```
Comment (PravinPK) [inline Swift/AEPSampleApp/MessagingView.swift:47]: shouldnt this be "Update Email on your AEPProfile"?

File: Swift/AEPSampleApp/MessagingView.swift [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            Edge.sendEvent(experienceEvent: experienceEvent) { (_: [EdgeEventHandle]) in
+                Log.debug(label: LOG_PREFIX, "Edge call is complete")
+            }
+        }
+    }
+}
+
+struct MessagingViewController_Previews: PreviewProvider {
+    static var previews: some View {
+        /*@START_MENU_TOKEN@*/Text("Hello, World!")/*@END_MENU_TOKEN@*/
```
Comment (PravinPK) [inline Swift/AEPSampleApp/MessagingView.swift:75]: "Hello world" :))

File: Documentation/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+## Configuration
+Before using messaging feature in the sample app you need to follow the below steps to configure it:
+
+1. Update the `Team` option in the `Siging & Capabilities` tab with the available Apple account. 
+1. Setup [Journey Optimizer](https://aep-sdks.gitbook.io/docs/beta/adobe-journey-optimizer)
+1. Update the `LAUNCH_ENVIRONMENT_FILE_ID` constant value in `AppDelegate.swift` file with the config app id of the property.
+    1. For more information on how to get the config app id from launch check this [document](https://experienceleague.adobe.com/docs/launch/using/publish/environments/environments.html?lang=en#mobile-configuration)
+1. Create an experience event schema and dataset for updating email id in the identity map. This step is required for merging the profile created with email id with the profile created with ECID.
+    1. Update the `EMAIL_UPDATE_DATASET` constant value in `AppDelegate.swift` file with the above created dataset id.
+1. After following the above steps `Run` the app on an iOS device (Simulators are not supported).
```
Comment (PravinPK) [inline Documentation/README.md:12]: I know all steps are super important. But they all cant be number 1

Comment (PravinPK) [inline Documentation/README.md:12]: `After following the above steps `Run` the app on an iOS device (Simulators are not supported).`

to 

`After following the above steps `Run` the app on an iOS device (Apple Push Notification cannot be delivered to a Simulator).`

File: Obj-C/AEPSampleAppObjC/AppDelegate.m [hunk 4]
```diff
@@ -14,7 +14,7 @@ @implementation AppDelegate
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
     
     [AEPMobileCore setLogLevel: AEPLogLevelTrace];
-    NSArray *extensionsToRegister = @[AEPMobileIdentity.class, AEPMobileLifecycle.class, AEPMobileSignal.class];
+    NSArray *extensionsToRegister = @[AEPMobileIdentity.class, AEPMobileLifecycle.class, AEPMobileSignal.class, AEPMobileMessaging.class];
```
Comment (PravinPK) [inline Obj-C/AEPSampleAppObjC/AppDelegate.m:17]: Are we not adding `setPushIdentifier` API to objectiveC app ?

File: Swift/AEPSampleApp/MessagingView.swift [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+import AEPCore
+import AEPEdge
+import AEPIdentity
+import AEPMessaging
+import AEPServices
+
+class MessagingViewController: UIHostingController<MessagingView> {}
+
+struct MessagingView: View {
+    @State private var personalisedData:String = ""
```
Comment (PravinPK) [inline Swift/AEPSampleApp/MessagingView.swift:22]: rename to personalEmail ?

File: Obj-C/AEPSampleAppObjC/AppDelegate.m (hunk 4 above)
Comment (ravjain-adb) [reply Obj-C/AEPSampleAppObjC/AppDelegate.m:17]: I did not update the objective c app

**Review by PravinPK — APPROVED** [review]

**Pull request merged**